src/Microsoft.Health.Dicom.Core/Features/Store/ValidationResultBuilder.cs

```diff
@@ -0,0 +1,104 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class ValidationResultBuilder : IValidationResultBuilder
+{
+    private readonly IList<string> _errorMessages;
+    private readonly IList<string> _warningMessages;
+
+    public ValidationResultBuilder()
+    {
+        _errorMessages = new List<string>();
+        _warningMessages = new List<string>();
+        WarningCodes = ValidationWarnings.None;
+    }
+
+    public bool HasWarnings
+    {
+        get { return _warningMessages.Count > 0; }
+    }
+
+    public bool HasErrors
+    {
+        get { return _errorMessages.Count > 0; }
+    }
+
+    public IEnumerable<string> Errors
+    {
+        get { return _errorMessages; }
+    }
+
+    public IEnumerable<string> Warnings
```

Comment:
What's the disctinction between `Warnings` and `WarningCodes`?

Reply:
Warnings is the "list of warning messages". The only reason I am keeping WarningCodes is to keep the current behavior. We just need either one.